pandas/core/frame.py

```diff
@@ -2590,9 +2590,24 @@ def assign(self, **kwargs):
 
         return data
 
-    def _sanitize_column(self, key, value):
-        # Need to make sure new columns (which go into the BlockManager as new
-        # blocks) are always copied
+    def _sanitize_column(self, key, value, broadcast=True):
+        """
+        Ensures new columns (which go into the BlockManager as new blocks) are
+        always copied and converted into an array.
+
+        Parameters
+        ----------
+        keys : object
+        value : int, Series, or array-like
+        broadcast : bool
+            Indicates whether all columns with the given key should be
```

Comment:
this is not obvious what you mean here. pls reword. (also this is not typically what ``broadcast`` means)